example_17/hardware/rrbot.cpp

```diff
@@ -0,0 +1,263 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ros2_control_demo_example_17/rrbot.hpp"
+
+#include <chrono>
+#include <cmath>
+#include <iomanip>
+#include <limits>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "diagnostic_updater/diagnostic_updater.hpp"
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace ros2_control_demo_example_17
+{
+hardware_interface::CallbackReturn RRBotSystemPositionOnlyHardware::on_init(
+  const hardware_interface::HardwareComponentInterfaceParams & params)
+{
+  if (
+    hardware_interface::SystemInterface::on_init(params) !=
+    hardware_interface::CallbackReturn::SUCCESS)
+  {
+    return hardware_interface::CallbackReturn::ERROR;
+  }
+
+  // Get Weak Pointer to Executor from HardwareComponentInterfaceParams
+  executor_ = params.executor;
+
+  // Ensure that the executor is available before creating the custom status node
+  if (auto locked_executor = executor_.lock())
+  {
+    std::string name_lower = info_.name;
+    std::transform(
+      name_lower.begin(), name_lower.end(), name_lower.begin(),
+      [](unsigned char c) { return std::tolower(c); });
+
+    std::string node_name = name_lower + "_custom_node";
+    custom_status_node_ = std::make_shared<rclcpp::Node>(node_name);
+
+    locked_executor->add_node(custom_status_node_->get_node_base_interface());
+  }
+
+  // BEGIN: This part here is for exemplary purposes - Please do not copy to your production code
+  hw_start_sec_ = stod(info_.hardware_parameters["example_param_hw_start_duration_sec"]);
```

Comment:
```suggestion
  // BEGIN: This part here is for exemplary purposes - Please do not copy to your production code
  hw_start_sec_ = stod(info_.hardware_parameters["example_param_hw_start_duration_sec"]);

  // Get Weak Pointer to Executor from HardwareComponentInterfaceParams
  executor_ = params.executor;

  // Ensure that the executor is available before creating the custom status node
  if (auto locked_executor = executor_.lock())
  {
    std::string name_lower = info_.name;
    std::transform(
      name_lower.begin(), name_lower.end(), name_lower.begin(),
      [](unsigned char c) { return std::tolower(c); });

    std::string node_name = name_lower + "_custom_node";
    custom_status_node_ = std::make_shared<rclcpp::Node>(node_name);

    locked_executor->add_node(custom_status_node_->get_node_base_interface());
  }
```
better move this above, as the custom executor is an example only